### ISSUE TITLE
How to get masked word prediction probabilities

### ISSUE THREAD

**Issue by Oxi84** [google-research/bert#608]
Original sentence: i love apples. there are a lot of fruits in the world that i like, but apples would be my favorite fruit. 
Masked sentence:  i love apples . there are a lot of fruits in the world that i [MASK] , but apples would be my favorite fruit .

When I run through the pytorch version of bert, I get the following representations of probabilities:

Best predicted word: ['love'] tensor(12.7276, grad_fn=<SelectBackward>)
Other words along with their probabilities:
['like'] tensor(10.2872, grad_fn=<SelectBackward>)
['miss'] tensor(8.8226, grad_fn=<SelectBackward>)
['know'] tensor(8.5971, grad_fn=<SelectBackward>)
['am'] tensor(7.9407, grad_fn=<SelectBackward>)
['hate'] tensor(7.9209, grad_fn=<SelectBackward>)
['mean'] tensor(7.8873, grad_fn=<SelectBackward>)
['enjoy'] tensor(7.8813, grad_fn=<SelectBackward>)
['want'] tensor(7.6885, grad_fn=<SelectBackward>)
['prefer'] tensor(7.5712, grad_fn=<SelectBackward>)

I am quite sure that this does not mean that probability for word "love" is proportional to 12.7276 and for word "like" is 10.2872.
I also know that the summ of all  func(this number) thought the whole vocabulary is 1. But I do not know what the func is?

Thanks

**Comment by Oxi84** [comment]
I see the that output function for BERT should be log_softmax, but IMO then all the values should be less than zero, because softwax is less that 1.

**Comment by nellymin** [comment]
I think that these values you have shown are the logits and they need to be passed through the log_softmax function. It would then produce the actual probabilities. As stated in the[ wikipedia page](https://en.wikipedia.org/wiki/Softmax_function)
`In mathematics, the softmax function, also known as softargmax[1] or normalized exponential function,[2]:198 is a function that takes as input a vector of K real numbers, and normalizes it into a probability distribution consisting of K probabilities. That is, prior to applying softmax, some vector components could be negative, or greater than one; and might not sum to 1; but after applying softmax, each component will be in the interval {\displaystyle (0,1)} (0,1), and the components will add up to 1, so that they can be interpreted as probabilities. ` 
so no wonders why they don't add up to 1

**Comment by Oxi84** [comment]
Thanks.
Do you mean softmax which is softmax_x1 = (e^x1/(e^x1 + e^x2 + ... e^Xn) or  log(softmax_x1)?
Thanks
